Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Handlers/PrimitiveManipulator.cs

```diff
@@ -0,0 +1,744 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Physics;
+using Microsoft.MixedReality.Toolkit.UI;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// This script allows for an object to be movable, scalable, and rotatable with one or two hands. 
+    /// You may also configure the script on only enable certain manipulations. The script works with 
+    /// both HoloLens' gesture input and immersive headset's motion controller input.
+    /// </summary>
+    [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/README_ManipulationHandler.html")]
+    public class PrimitiveManipulator : MonoBehaviour, IMixedRealityPointerHandler, IMixedRealityFocusChangedHandler
+    {
+        #region Public Enums
+        [System.Flags]
+        public enum HandMovementType
+        {
+            OneHanded = 1 << 0,
+            TwoHanded = 1 << 1,
+        }
+        [System.Flags]
+        public enum TwoHandedManipulation
+        {
+            Move = 1 << 0,
+            Rotate = 1 << 1,
+            Scale = 1 << 2
+        };
+        public enum RotateInOneHandType
+        {
+            MaintainRotationToUser,
+            GravityAlignedMaintainRotationToUser,
+            FaceUser,
+            FaceAwayFromUser,
+            MaintainOriginalRotation,
+            RotateAboutGrabPoint
```

Comment:
Looks like you removed RotateAboutObjectCenter, but actually I found that very useful for rotating things with one hand at a distance. I noticed that the behavior is broken in the 2.0 release, and filed #5957. Maybe you can fix #5957 in the new manipulationhandler, and we can discuss whether #5957 needs to be fixed for the old one (I posted a workaround that people can do in the mean time).

Reply:
Note that doing #5957 core idea is for far manipulation you need to use the grip pose instead of the pointer pose -- the grip pose rotates with the palm, but the pointer pose uses the hand ray. For motion controllers it's similar -- grip pose is the rotation of the 6dof where pointer rotates based on where the 6dof is 'pointing' (not the same as grip pose).

Reply:
Ah ok, I removed because I had only observed the broken behaviour. I'll fix and add this option back.